### DISCUSSION TITLE
Welcome to batch_PDF_compressor Discussions!

### DISCUSSION THREAD

**Discussion by sergeiown** [sergeiown/batch_PDF_compressor#4] [Announcements]
## 👋 Welcome!

**Comment by Megazoi** [comment]
hello dear Sergei

can you help me with your program ?

**Reply by sergeiown** [reply]
Hello, in what way exactly?

**Reply by Megazoi** [reply]
I have Win7 64x system and try to run your application.

Gostscript was downloaded from here https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs10021/gs10021w64.exe

and installed

Try to run batch_pdf_compressor.exe and just get the message "The system cannot write to the specified device" in cmd window.

What I am doing wrong ?

How does your application have to work ?

**Reply by sergeiown** [reply]
This may be due to access restrictions on the device or directory you are trying to write to. Make sure you have the appropriate access rights to the device or directory. This usually happens when you use an account with low access rights (not administrator). Windows 7 is unlikely to have any effect on the process.

**Comment by sergeiown** [comment]
Basically, if it's that important to you... You can simply use the command line: gswin64c.exe -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/printer -dNOPAUSE -dQUIET -dBATCH -sOutputFile=[output file] [input file] to check if Ghostscript is working normally on your system.

**Reply by Megazoi** [reply]
yeah. this command line is working fine. I tried it with some pdf file and it was compressed more than double  !!!  very nice !!!  

But batch_pdf_compressor.exe is still not working ... same error... I have a administrator rights for my account and run it from c:\users\user\documents\batch_pdf_compressor\ directory. My account has full rights for this directory.

I tried to run it under local administrator account "Administrator" but got same error ...

very strange ...

**Comment by Megazoi** [comment]
where should be located input pdf filer for this program ? maybe in some special directory ?

I have a batch_pdf_compressor dir after installation and following files inside ---

unisnt000.dat
uninst000.exe
batch_pdf_compressor.ico
batch_pdf_compressor.exe

how does this programm work ? where does it find the input pdf files and where save output files after compression ?

**Reply by sergeiown** [reply]
You just need to execute the exe file and get an invitation to select the directory with the pdf files, as shown in the screenshot below:![image](https://github.com/sergeiown/batch_PDF_compressor/assets/112722061/2025bd80-ac6a-4ff2-b642-a16f6099d417)
You choose the input directory, and the output files are saved in the same location. Very, very simple...

**Reply by sergeiown** [reply]
exe in this case is just a container for bat scripts that you see in the repository. You can use them, it's the same thing.
